lib/plugins/boss_bar.js

```diff
@@ -1,47 +1,43 @@
 module.exports = inject
 
 function inject (bot, { version }) {
-  const ChatMessage = require('../chat_message')(version)
-
-  const colors = [ 'pink', 'blue', 'red', 'green', 'yellow', 'purple', 'white' ]
-  const divisions = [ 0, 6, 10, 12, 20 ]
+  const BossBar = require('../bossbar')(version)
   const bars = {}
 
   bot._client.on('boss_bar', (packet) => {
     if (packet.action === 0) {
-      bars[packet.entityUUID] = {
-        entityUUID: packet.entityUUID,
-        title: new ChatMessage(JSON.parse(packet.title)),
-        health: packet.health,
-        dividers: divisions[packet.dividers],
-        colors: colors[packet.color],
-        shouldDarkenSky: packet.flags & 0x1,
-        isDragonBar: packet.flags & 0x2,
-        createFog: packet.flags & 0x4
-      }
+      bars[packet.entityUUID] = new BossBar(
+        packet.entityUUID,
+        packet.title,
+        packet.health,
+        packet.dividers,
+        packet.color,
+        packet.flags
+      )
 
       bot.emit('bossBarCreated', bars[packet.entityUUID])
     } else if (packet.action === 1) {
       bot.emit('bossBarDeleted', bars[packet.entityUUID])
       delete bars[packet.entityUUID]
     } else {
+      if (!(packet.entityUUID in bars)) {
```

Comment:
This is the code that fixes the error / crash when a bar update is received without it being registered.